### PR TITLE
GameDB: Vampire Night fixes and code formatting.

### PR DESCRIPTION
**Pull request by atomic83GitHub**
A short version of the closed pull request with some new adds, this was made in order to save the time of the devs and for avoid users to manualy add these folowing hacks.

All games have been tested, can be merge.

- [x] Rebase to fix merge conflict.
- [x] Explanation in commit message why Vampire Night EETiminghack was removed.
- [x] Clean up all new comments by starting with a capital letter and ending with a point.

### CODE REVIEW

**Comment by tony971** [comment]
I can confirm that Need for Speed: Carbon works with the changes. Don't know about the rest.

File: bin/GameIndex.dbf [hunk 1]
```diff
@@ -1263,7 +1278,8 @@ Serial = SCES-50354
 Name   = Klonoa 2 - Lunatea's Veil
 Region = PAL-M5
 Compat = 5
-eeClampMode = 3		//Objects needed appear in wrong places without it
+eeClampMode = 3 //Objects needed appear in wrong places without it
```
Comment (lightningterror) [inline bin/GameIndex.dbf]: Small nitpick, maybe add space between // and Objects so it's consistent with the other changes.
Could do the rest with the others that need it as well.

**Comment by ElementUranium** [comment]
delete 

image

**Comment by atomic83GitHub** [comment]
I remove untested games, you can safely merge the pull request.

File: bin/GameIndex.dbf [hunk 2]
```diff
@@ -923,6 +923,7 @@ Region = NTSC-Unk
 Serial = SCAJ-20181
 Name   = Minna no Tennis
 Region = NTSC-Unk
+eeRoundMode = 0
```
Comment (ssakash) [inline bin/GameIndex.dbf]: It would be nice to add a comment on why this rounding mode is necessary.

File: bin/GameIndex.dbf [hunk 3]
```diff
@@ -984,6 +985,7 @@ VuAddSubHack = 1
 Serial = SCAJ-20198
 Name   = Everybody's Tennis [PlayStation 2 The Best]
 Region = NTSC-Unk
+eeRoundMode = 0
```
Comment (ssakash) [inline bin/GameIndex.dbf]: ditto

File: bin/GameIndex.dbf [hunk 4]
```diff
@@ -2244,6 +2248,8 @@ Serial = SCES-53409
 Name   = Sly 3
 Region = PAL-M11
 Compat = 5
+vuRoundMode = 0
```
Comment (ssakash) [inline bin/GameIndex.dbf]: ditto

File: bin/GameIndex.dbf [hunk 5]
```diff
@@ -2585,6 +2591,7 @@ Region = PAL-Unk
 Serial = SCES-54535
 Name   = Everybody's Tennis
 Region = PAL-M11
+eeRoundMode = 0
```
Comment (ssakash) [inline bin/GameIndex.dbf]: ditto and same for all other options without a comment in it, it would take me eons to add a ditto comment for every line. xD

**Comment by atomic83GitHub** [comment]
Done.

**Comment by pgert** [comment]
"These fix are needed to display graphics instead of a black screen."
It's eighter "This fix is" (for one fix), or "These fixes are" for several.

**Comment by atomic83GitHub** [comment]
I have update my PR with some arapapa and @yxmline's request.

**Comment by MrCK1** [comment]
Did you test Men in Black yourself or did you go off of the data in the wiki? The test case there is ftom 0.9.9, so the hack may not be needed anymore.

Also, don't forget to squash your commits

**Comment by atomic83GitHub** [comment]
No, I don't test it but it was suggested by arapapa on the forum, it would be better to ask this to him :).

**Comment by atomic83GitHub** [comment]
So I have 2 PCPX titles to put:

Serial = PCPX-96660
Name   = Tourist Trophy [Trial]  
Region = NTSC-J  
--------------------------------------------- 
Serial = PCPX-98042 
Name   = Hot Shots Tennis |Demo, Taikenban] 
Region = NTSC-J 
eeRoundMode = 0 // fix some missing 2d elements 
--------------------------------------------- 

@lightningterror where I should put them?

**Comment by lightningterror** [comment]
Put them both under PCPX-96649.

**Comment by atomic83GitHub** [comment]
Done.

**Comment by MrCK1** [comment]
What's the status on this?

**Comment by FlatOutPS2** [comment]
> What's the status on this?

The `Compat = 5`'s should be removed. And then all the changes need to be confirmed to be needed.

The Elvandia Story one can definitely not be included in the index as its a wrong interpretation from an old tester who you can't rely on for accurate information.

File: bin/GameIndex.dbf [hunk 6]
```diff
@@ -18384,6 +18430,11 @@ Name   = NanoBreaker
 Region = NTSC-J
 Compat = 5
 ---------------------------------------------
+Serial = SLKA-25264
+Name   = Full Spectrum Warrior
+Region = NTSC-K
+EETimingHack = 1 // flickery textures
```
Comment (FlatOutPS2) [inline bin/GameIndex.dbf]: // fixes broken effects

File: bin/GameIndex.dbf [hunk 7]
```diff
@@ -1616,6 +1626,9 @@ Region = PAL-M5
 Serial = SCES-51176
 Name   = Disney's Treasure Planet
 Region = PAL-M4
+EETimingHack = 1 // Fixes hang before going ingame
+eeRoundMode = 0 // Fixes characters flying into sky
+eeClampMode = 3 // Required to load areas correctly
```
Comment (FlatOutPS2) [inline bin/GameIndex.dbf]: I'm not sure these are all still required, there might be differences between regions. Needs confirmation.

**Comment by pgert** [comment]
@atomic83GitHub
Typo-error:
Serial = PCPX-98042
Name = Hot Shots Tennis |Demo, Taikenban]

Should be:
Serial = PCPX-98042
Name = Hot Shots Tennis [Demo, Taikenban]
(the left bracket)

File: bin/GameIndex.dbf (hunk 7 above)
Comment (atomic83GitHub) [reply bin/GameIndex.dbf]: Yes, I will keep this for now, waiting for test :).

File: bin/GameIndex.dbf [hunk 8]
```diff
@@ -168,6 +168,15 @@ vuClampMode = 2 // Text in GT mode works
...
 //[/patches]
 ---------------------------------------------
+Serial = PCPX-96660
+Name = Tourist Trophy [Trial]
+Region = NTSC-J
+---------------------------------------------
+Serial = PCPX-98042
+Name = Hot Shots Tennis [Demo, Taikenban]
+Region = NTSC-J
+eeRoundMode = 0 // fix some missing 2d elements
```
Comment (FlatOutPS2) [inline bin/GameIndex.dbf]: `Fixes missing HUD elements.`

File: bin/GameIndex.dbf [hunk 9]
```diff
@@ -15841,7 +15872,7 @@ Serial = SLES-54308
 Name   = Phantasy Star Universe
 Region = PAL-M3
 Compat = 5
-eeRoundMode = 0
+eeRoundMode = 0 // Fix for ennemies not moving
```
Comment (FlatOutPS2) [inline bin/GameIndex.dbf]: `Fixes enemies not moving.`

@Ryudo300 mentioned an issue with Chapter 7's boss that was fixed by setting this to positive. Are you sure this is the best setting?

File: bin/GameIndex.dbf [hunk 10]
```diff
@@ -26947,6 +27066,12 @@ Serial = SLPM-66585
 Name   = Elvandia Story
 Region = NTSC-J
 Compat = 5
+// One of these fixes are needed to display graphics instead of a black screen.
```
Comment (FlatOutPS2) [inline bin/GameIndex.dbf]: Unless you found the correct fix, you should remove this.

**Comment by FlatOutPS2** [comment]
Other changes that should be done before merging:
[ ] - Rebase to fix merge conflict.
[ ] - Explanation in commit message why Vampire Night EETiminghack was removed.
[ ] - Clean up all new comments by starting with a capital letter and ending with a point.

File: bin/GameIndex.dbf (hunk 9 above)
Comment (atomic83GitHub) [reply bin/GameIndex.dbf]: Rounding to positive or nearest does the same thing (solve the entire chapter 7 (Boss include)).

Comment (FlatOutPS2) [reply bin/GameIndex.dbf]: Yes, but it mentions an issue with the boss remaining.

**Comment by pgert** [comment]
"fix hangs when entering in the garden" should be "Fixes hang when entering the garden.".

"For freezes in some scene transitions." should be "Fixes freezes in some scene transitions.", or "Prevents freezes in some scene transitions.".

**Comment by atomic83GitHub** [comment]
Done.

**Comment by lightningterror** [comment]
`Game DB changes by atomic83`
It's not really needed to add your name on the commit name since the author is shown on both github and the buildbot.
You can name it to something like **GameDB: Vampire Night fixes and code formatting.**
Also the conflict still needs to be resolved.

File: bin/GameIndex.dbf [hunk 11]
```diff
@@ -2244,6 +2254,8 @@ Serial = SCES-53409
 Name   = Sly 3
 Region = PAL-M11
 Compat = 5
+vuRoundMode = 0 // Fix physics issue in some levels.
```
Comment (FlatOutPS2) [inline bin/GameIndex.dbf]: You need to be more specific here. Physics issue and some level aren't very clear. I know these settings solve a bomb mission in Episode 5, but is it required for other levels too?

File: bin/GameIndex.dbf [hunk 12]
```diff
@@ -15841,7 +15875,7 @@ Serial = SLES-54308
 Name   = Phantasy Star Universe
 Region = PAL-M3
 Compat = 5
-eeRoundMode = 0
+eeRoundMode = 2 // Fix for ennemies not moving.
```
Comment (FlatOutPS2) [inline bin/GameIndex.dbf]: I assume this only affects battles, so `Fixes enemies not moving during battles.`

**Comment by lightningterror** [comment]
The commit name still isn't updated.

File: bin/GameIndex.dbf [hunk 13]
```diff
@@ -38721,6 +38886,9 @@ Region = NTSC-U
 Serial = SLUS-21054
 Name   = Disney's Classics - Treasure Planet
 Region = NTSC-U
+EETimingHack = 1 // Fixes hang before going ingame
+eeRoundMode = 0 // Fixes characters flying into sky
+eeClampMode = 3 // Required to load areas correctly
```
Comment (lightningterror) [inline bin/GameIndex.dbf]: Missing periods here on the 3 rows.

File: bin/GameIndex.dbf [hunk 14]
```diff
@@ -2244,6 +2254,8 @@ Serial = SCES-53409
 Name   = Sly 3
 Region = PAL-M11
 Compat = 5
+vuRoundMode = 0 // Fix issues with camera in some levels.
```
Comment (FlatOutPS2) [inline bin/GameIndex.dbf]: What camera issues?

File: bin/GameIndex.dbf [hunk 15]
```diff
@@ -2244,6 +2254,8 @@ Serial = SCES-53409
...
+vuClampMode = 2 // Fix game engine issues like the bombs not exploding and the inability to take a ladder throughout the game.
```
Comment (FlatOutPS2) [inline bin/GameIndex.dbf]: `Fixes not being able to climb a ladder and fixes Bentley's bombs not working in the Episode 5 mission "Dynamic Duo".`

I assume you meant climbing a ladder, not literally taking a ladder.

**Comment by atomic83GitHub** [comment]
I am currently rebasing this PR, this should be fix soon.

**Comment by lightningterror** [comment]
The conflict still remains.

**Comment by atomic83GitHub** [comment]
Yes, I am currently trying to find what cause it.

**Comment by lightningterror** [comment]
I can't check which lines cause the conflicts since I don't have write access to the master branch.

**Comment by willkuer** [comment]
You don't need write access to master. Just checkout his branch/commit and then `git rebase -i pcsx2/master`, do rebase, upon merge conflicts investigate and solve. Then `git add` the modified files and `git rebase --continue`.

However I think he should do it by himself to learn the git workflow. It's basically used everywhere nowadays.

**Comment by atomic83GitHub** [comment]
Is it better now?

**Comment by atomic83GitHub** [comment]
I did that but there is apparently no conflicts.

image

**Comment by willkuer** [comment]
Depends on your tracking. Did you first fork and then branch your pr from that fork? Then I'd suggest to reference pcsx2/master and not origin/master as the latter is probably your fork (which might not have any merge conflicts). 

Otherwise you can also adjust the tracking by - i think - `git branch -u pcsx2/master` so that 'git rebase -i` will suffice.

**Comment by willkuer** [comment]
If you fix the tracking and do a `git status` you should get the measage `your branch is one commit ahead of pcsx2/master` or something similar. Your screenshot already shows that your tracking is not optimal. And I think you are missing the latest updates of pcsx2/master.

**Comment by atomic83GitHub** [comment]
When I am trying to update the branch, I am getting this error:
image

And git status show this:
image

**Comment by lightningterror** [comment]
For some reason I can't see where the conflict starts and ends.

It might be a little bit of work but maybe you can solve it this way:
Copy paste the GameDB from master and then re add the changes.

Some of the newer gamedb prs caused the conflicts (shared lines of codes somewhere).

**Comment by atomic83GitHub** [comment]
Done

File: bin/GameIndex.dbf [hunk 16]
```diff
@@ -18573,6 +18800,16 @@ Region = NTSC-K
...
+Serial = SLKA-25372 
+Name   = Black 
+Region = NTSC-K 
+vuClampMode = 0 // Fix broken graphics ingame. 
+--------------------------------------------- 
+Serial = SLKA-25374
+Name   = Fifa Street 2
+Region = NTSC-K
+GIFFIFOHack = 1 // Fix flag corruptions
+---------------------------------------------
```
Comment (lightningterror) [inline bin/GameIndex.dbf]: Okay this is the merge conflict probably, can you exclude/remove the two added serials ?

Comment (lightningterror) [reply bin/GameIndex.dbf]: Serial = SLKA-25372 
Serial = SLKA-25374

**Comment by lightningterror** [comment]
Nice that was correct then. So there's still a bit cleaning up to do before merge.

File: bin/GameIndex.dbf [hunk 17]
```diff
@@ -43332,6 +43853,7 @@ Region = NTSC-U
 Serial = SLUS-29153
 Name   = Burnout Revenge [Demo]
 Region = NTSC-U
+vuClampMode = 0 //Fixes buggy lighting on certain objects
```
Comment (lightningterror) [inline bin/GameIndex.dbf]: `//Fixes` Space in between and period (dot) at the end of the comment.

File: bin/GameIndex.dbf [hunk 18]
```diff
@@ -43239,6 +43759,7 @@ Region = NTSC-U
 Serial = SLUS-29113
 Name   = Burnout 3 [Demo]
 Region = NTSC-U
+vuClampMode = 0 //Fixes buggy lighting on certain objects
```
Comment (lightningterror) [inline bin/GameIndex.dbf]: Same like before.

**Comment by lightningterror** [comment]
There are a bit more with period/dots missing, double check and correct them.

File: bin/GameIndex.dbf [hunk 19]
```diff
@@ -6476,6 +6536,7 @@ Region = PAL-E
 Serial = SLED-53953
 Name   = FIFA Street 2 [Demo]
 Region = PAL-E
+GIFFIFOHack = 1 // Fix flag corruptions
```
Comment (lightningterror) [inline bin/GameIndex.dbf]: Did you actually test Fifa Street 2? I belive the flags corruption can be fixed with blending half, the other half a shader (8bit texture conversion) fixes that in OpenGL.

Comment (lightningterror) [reply bin/GameIndex.dbf]: Ok, can you upload a gs dump ? I want to be sure before this is merged.

Comment (lightningterror) [reply bin/GameIndex.dbf]: Actually better a block dump with a save state.

File: bin/GameIndex.dbf [hunk 20]
```diff
@@ -37961,9 +38433,10 @@ Region = NTSC-U
 Compat = 5
 ---------------------------------------------
 Serial = SLUS-20909
-Name   = Crash Bandicoot Twinsanity
+Name   = Crash Twinsanity
 Region = NTSC-U
 Compat = 5
+XgKickHack = 1 // Fixes bad geometry.
```
Comment (lightningterror) [inline bin/GameIndex.dbf]: Better remove these changes as well, @MrCK1 added the changes a week ago.

**Comment by pgert** [comment]
"Fix reverse control and boss in some places" should be "Fixes reverse control and boss in some places." (Fixes and Period)

"Fix issues with camera became stuck in a wall or under Sly some levels." should be
 "Fixes issues with camera getting stuck in a wall or under Sly in some levels."

"//Fixes buggy lighting on certain objects" should be "// Fixes buggy lighting on certain objects." (Space and Period)

""Fix flag corruptions" should be ""Fixes flag corruptions." (Fixes and Period)

"fix missing/grey texture or geometry ingame" should be "Fixes missing/grey texture or geometry ingame." (Fixes and Period)

"fixes some lighting/shadow artefacts (most light sources like the Flashlight are unaffected)" should be
 "Fixes some lighting/shadow artefacts (most light sources like the Flashlight are unaffected)." (Fixes and Period)

"avoids obscured FMVs due to GS upscaling lines and memory wrapping" should be
 "Avoids obscured FMVs due to GS upscaling lines and memory wrapping."

"fix SPS with water in some places" should be "Fixes SPS with water in some places." (Fixes and Period)

"Fix jump issue" should be "Fixes jump issue." (Fixes and Period)

"Fix camera issue" should be "Fixes camera issue." (Fixes and Period)

"Fix target loss issue" should be "Fixes target loss issue." (Fixes and Period)

"fix hang at loading screen" should be "Fixes hang at loading screen." (Fixes and Period)

"Avoid hanging at the "press start" screen" should be "Avoids hanging at the "press start" screen." (Avoids and Period)

"Avoids", not "Avoid":
"Avoid hanging at loading screen." should be "Avoids hanging at loading screen."
"Avoid hang at start." should be "Avoids hang at start."
"Avoid hang at loading." should be "Avoid hang at loading."

Period:
"Fixes crashes"
"Fixes transparency issues"
"Objects appear in wrong places without it"
"Fixes water reflection"
"Fixes "Fall through floor" bug when aproaching chests"
"Fixes missing light shafts under lamps"
"Fixes hang before going ingame"
"Fixes characters flying into sky"
"Required to load areas correctly"
"Fixes cut-off numbers and restores missing whitespace inside combo meter"

"Fixes", not "Fix":
"Fix game engine issues like the bombs not exploding and the inability to climb a ladder throughout the game."
"Fix Z-fighting at specific angles"
"Fix Abnormal AI behavior"
"Fix boot hanging."
"Fix hang at loading screen."
"Fix initial hang."
"Fix hang at Level 4."
"Fix hang at shipping shed."
"Fix hanging problem."

"Harry Potter to Himitsu no Heya(Coca Cola Original version)" should be
 "Harry Potter to Himitsu no Heya [Demo - Coca Cola Original Version]" don't you think?

"Dvd version." should be "DVD version."
"Cd version." should be "CD version."

"comment=This disc has the same crc as SLES-52895." should be "comment=- This disc has the same CRC as SLUS-20904, the NTSC-U disc."
"comment=This disc has the same CRC as SLES-52812, the PAL-E disc." should be "comment=- This disc has the same CRC as SLES-52812, the PAL-E disc."

**Comment by lightningterror** [comment]
RIP, looks like the last PR I merged with GameDB changes caused more conflicts on this one.

**Comment by atomic83GitHub** [comment]
I don't finish yet but I merge my current work for @lightningterror.

**Comment by lightningterror** [comment]
Search for 

`<<<<<<< HEAD`

That's where the conflicts begin

ends with

`>>>>>>>`

There are a lot now in the file.

[GameIndex.zip](https://github.com/PCSX2/pcsx2/files/1972597/GameIndex.zip)

**Comment by lightningterror** [comment]
Ok the flag corruption is in SW mode as well and I see why it's needed.

Can you fix the conflicts and rebase ?

**Comment by atomic83GitHub** [comment]
@lightningterror I fixed it internally but I am still adding/edit some fixes before pushing the commit.

**Comment by lightningterror** [comment]
Did you take a look at the file I sent ? The conflicts haven't changed.

**Comment by atomic83GitHub** [comment]
Yes, I had fix all conflicts in your file one by one but I don't know what's going on here, I will redo this a second time to see if it will change something.

**Comment by atomic83GitHub** [comment]
I'll do like the last time, take an updated version of the GameDB and redo the neccessary changes

**Comment by lightningterror** [comment]
Okay that worked, some lines need a bit cleaning up tho but I'm not sure if that will create conflicts or not.

There are some spaces that need to be added after // and some whitespaces that need removing before //.
There are some periods (dots) missing from some sentences.

I backed up this db just in case more conflicts arise.

**Comment by atomic83GitHub** [comment]
Understood :).

**Comment by atomic83GitHub** [comment]
@lightningterror can you tell me where is the conflict this time?

**Comment by lightningterror** [comment]
[GameIndex.zip](https://github.com/PCSX2/pcsx2/files/1996824/GameIndex.zip)

**Comment by lightningterror** [comment]
Search <<<<<<< HEAD

**Comment by turtleli** [comment]
It would be better to rebase and fix the merge conflicts than to not include stuff that'll cause a merge conflict.

@atomic83GitHub What is the output of `git remote -v`?

**Comment by atomic83GitHub** [comment]
@turtleli It's this:

capture6

**Comment by turtleli** [comment]
```
git fetch upstream
git rebase upstream/master
```
Fix the merge conflicts, then
```
git rebase --continue
```

**Comment by atomic83GitHub** [comment]
So I will do the following: integrate the necessary fixes and trying to do a proper cleanup in a separate PR later.

**Comment by atomic83GitHub** [comment]
Done and it work, now I will have to integrate the necessary fixes and trying to do a proper cleanup in a separate PR.

**Comment by atomic83GitHub** [comment]
Seems good now :).

File: bin/GameIndex.dbf [hunk 21]
```diff
@@ -36756,9 +36775,8 @@ Name   = Harry Potter and The Chamber of Secrets
 Region = NTSC-U
 Compat = 2
 [patches = C5473413]
-	// This disc has the same CRC as SLES-51192, the PAL-E disc.
```
Comment (turtleli) [inline bin/GameIndex.dbf]: Ditto.

**Comment by pgert** [comment]
"Rearranging COP2 ops to avoids macflag bad stuff" ought to be "Rearranging COP2 ops so as to avoid macflag bad stuff".

**Comment by pgert** [comment]
... "so as to avoid", not "so as to avoids".

File: bin/GameIndex.dbf [hunk 22]
```diff
@@ -8376,39 +8384,39 @@ Region = PAL-E
 Compat = 5
 vuRoundMode = 0
 vuClampMode = 3
-eeClampMode = 2	// fix missing/grey texture or geometry ingame
+eeClampMode = 2	// Fixes missing/grey texture or geometry ingame.
```
Comment (lightningterror) [inline bin/GameIndex.dbf]: Maybe the spacing on these comments could be adjusted.

There are others as well.

**Comment by lightningterror** [comment]
I prefer if **_GameDB:_** was added in the commit title.

File: bin/GameIndex.dbf [hunk 23]
```diff
@@ -1181,12 +1182,10 @@ Name   = Dead or Alive 2 - Hardcore
...
 [patches = 7A51F86E]
-	comment=Patch by Prafull and kozarovv
-	// Fixes game hanging on boot.
+	comment=Patch by Prafull
+	// Fixes game hanging on boot, but removes all sound effects.
 	patch=0,EE,002b4c44,word,24060000
-	patch=1,IOP,001D77E4,word,27C40010
-	patch=1,IOP,001D77E8,word,0C0032B7
-	patch=1,IOP,001D77EC,word,00000000
+	patch=0,EE,002b4dc0,word,10000014
```
Comment (lightningterror) [inline bin/GameIndex.dbf]: Can you also revert these changes for SCES-50003 (Dead or Alive 2).

I updated the patch a few days ago.
